src/hooks/room/useGuestAccessInformation.ts

```diff
@@ -52,7 +52,10 @@ export const useGuestAccessInformation = (room: Room): GuestAccessInformation =>
         [canChangeJoinRule, isRoomJoinable, guestSpaUrl],
     );
 
-    const isRoomJoinableFunction = (): boolean =>
-        room.getJoinRule() === JoinRule.Public || (joinRule === JoinRule.Knock && room.canInvite(room.myUserId));
+    const isRoomJoinableFunction = (): boolean => {
+        const joinRule = room.getJoinRule();
```

Comment:
aren't you now shadowing joinRule? If it's the same thing, shouldn't this be a callback that depends on the value of joinRule?

Reply:
AIUI _not_ shadowing the `joinRule` is what caused the bug this PR aims to fix.

Reply:
yes we need this to be computed on the fly.
This method is called from a callback that gets created when we show the first modal (asking to change permission rules)

Reply:
right, but if it's a different variable, it should generally have a different name rather than shadowing a variable in a higher scope. Wouldn't having it as a dependency make react do the right thing?

Reply:
> right, but if it's a different variable, it should generally have a different name rather than shadowing a variable in a higher scope. 

That is 100% correct. I see your point now.

I think having the stateVar here does not work because of this
```
    const shareClick = useCallback(() => {
        if (isRoomJoinable()) {
            showLinkModal();
        } else {
            // the room needs to be set to public or knock to generate a link
            Modal.createDialog(JoinRuleDialog, {
                room,
                // If the user cannot invite the Knocking is not given as an option.
                canInvite,
            }).finished.then(() => {
                if (isRoomJoinable()) showLinkModal();
            });
        }
    }, [isRoomJoinable, showLinkModal, room, canInvite]);

```
this line `  if (isRoomJoinable()) showLinkModal();` would not update the state variable even if it is a dependency since it would capture the variable while it is awaiting.